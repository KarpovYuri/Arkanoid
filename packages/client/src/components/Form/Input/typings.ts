export type InputProps = {
  inputName: string;
  placeholder: string;
  type: string;
  title: string;
  pageType: string;
  errorMessage: string;
  required?: boolean;
  minLength?: number;
  maxLength?: number;
  pattern?: string;
  handleValidate: (evt: React.ChangeEvent<HTMLInputElement>) => void;
  inputValue?: string | null;
};
