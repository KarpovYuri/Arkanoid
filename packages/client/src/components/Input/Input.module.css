.input {
  padding: 26px 30px;
  margin-top: 16px;
  font-size: 20px;
  line-height: 24px;
  background: none;
  border: 2px solid var(--border-color);
  border-radius: 8px;
  color: var(--font-primary-color);
}

.input:focus {
  outline: 2px solid var(--accent-color);
}

.input::placeholder {
  color: var(--font-secondary-color);
  opacity: 1;
  transition: opacity 0.25s ease;
}

.input:focus::placeholder {
  opacity: 0;
  transition: opacity 0.25s ease;
}
