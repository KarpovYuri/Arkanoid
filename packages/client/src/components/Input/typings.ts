export type TInput = {
  value?: string;
  classname?: string;
  placeholder: string;
  name: string;
  type: string;
  onChange?: (e: React.FormEvent<HTMLInputElement>) => void;
};
