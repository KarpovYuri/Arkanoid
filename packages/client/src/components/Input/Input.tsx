import classNames from 'classnames';
import styles from './Input.module.css';
import { TInput } from './typings';

export function Input(props: TInput) {
  const { classname, placeholder, name, type, onChange } = props;
  const classes = classNames(styles.input, classname);

  return (
    <input
      onChange={onChange}
      type={type}
      className={classes}
      name={name}
      placeholder={placeholder}
    />
  );
}
