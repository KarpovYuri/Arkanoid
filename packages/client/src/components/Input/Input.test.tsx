import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Input } from './Input';

describe('Input component', () => {
  test('renders correctly', () => {
    render(<Input placeholder="test placeholder" name="test" type="text" />);

    const input = screen.getByPlaceholderText(/test placeholder/i);

    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });
});
