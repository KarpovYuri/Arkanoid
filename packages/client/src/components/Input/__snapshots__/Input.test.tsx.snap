// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<input
  class=""
  name="test"
  placeholder="test placeholder"
  type="text"
/>
`;
